### PR TITLE
Feature/26753/wallet web socket

### PR DESCRIPTION
**Pull request by AlexanderBess**

### CODE REVIEW

File: pages/wallet.vue [hunk 1]
```diff
@@ -154,6 +154,7 @@
 <script>
 import { mapGetters } from 'vuex';
 import BigNumber from 'bignumber.js';
+import moment from 'moment';
```
Comment (matykoks) [inline pages/wallet.vue:157]: use this.$moment

File: store/wallet/actions.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ import {
   getTransferFeeData,
   getContractFeeData,
   getIsWalletConnected,
-  sendWalletTransaction,
+  sendWalletTransaction, getProvider,
```
Comment (matykoks) [inline store/wallet/actions.js:15]: del getProvider

File: pages/wallet.vue [hunk 3]
```diff
@@ -259,8 +260,17 @@ export default {
     if (!this.isWalletConnected) return;
     const i = this.tokenSymbolsDd.indexOf(this.selectedToken);
     this.ddValue = i >= 0 && i < this.tokenSymbolsDd.length ? i : 1;
+    await this.$store.dispatch('wallet/subscribeToWalletEvents', {
+      address: this.convertToBech32('eth', this.userWalletAddress),
```
Comment (matykoks) [inline pages/wallet.vue:264]: ethm

File: store/wallet/actions.js [hunk 4]
```diff
@@ -435,4 +434,54 @@ export default {
       return error();
     }
   },
+  async subscribeToWalletEvents({
+    commit, dispatch, rootGetters, getters,
+  }, payload) {
+    try {
+      this.$wsNode.connect(null);
+      this.connection = new WebSocket('wss://dev-node-nyc3.workquest.co/tendermint-rpc/websocket');
```
Comment (matykoks) [inline store/wallet/actions.js:442]: variable?

Comment (matykoks) [reply store/wallet/actions.js:442]: get node address from env

File: store/wallet/actions.js [hunk 5]
```diff
@@ -435,4 +434,54 @@ export default {
...
+            status: true,
+            value: events['ethereum_tx.amount'][0],
+            transaction_fee: +(events['tx.fee'][0].split('a')[0]),
+            from_address_hash: { hex: events['message.sender'][3] },
+            to_address_hash: { hex: recipient },
+          });
+          await payload.updateWalletData();
+          commit('setTransactions', transactions);
+          let transactionsCount = JSON.parse(JSON.stringify(getters.getTransactionsCount));
+          commit('setTransactionsCount', transactionsCount += 1);
```
Comment (matykoks) [inline store/wallet/actions.js:475]: ` commit('setTransactionsCount', getters.getTransactionsCount + 1);`

File: store/wallet/actions.js [hunk 6]
```diff
@@ -435,4 +434,54 @@ export default {
...
+        }
+      };
+      return true;
+    } catch (err) {
+      console.error(err);
+      return error();
+    }
+  },
+  async unsubscribeToWalletEvents({ commit, dispatch }) {
+    this.$wsNode.disconnect();
```
Comment (matykoks) [inline store/wallet/actions.js:485]: need disconnect  from this.connection

File: plugins/ws.js [hunk 7]
```diff
@@ -11,6 +11,9 @@ export default async ({ store, $axios }, inject) => {
       {
         name: 'wsChatActions', url: process.env.WS_CHAT_ACTIONS_URL, verboseName: 'chatActionsConnection',
       },
+      {
+        name: 'wsNode', url: process.env.WS_WQ_PROVIDER, verboseName: 'node',
```
Comment (matykoks) [inline plugins/ws.js:15]: del

File: pages/wallet.vue [hunk 8]
```diff
@@ -259,8 +260,17 @@ export default {
...
+      hexAddress: this.userWalletAddress,
+      date: moment(),
+      updateWalletData: async () => await this.loadData(),
+    });
     await this.loadData();
   },
+  async beforeDestroy() {
+    await this.$store.dispatch('wallet/unsubscribeToWalletEvents');
```
Comment (matykoks) [inline pages/wallet.vue:272]: use simple names

Comment (AlexanderBess) [reply pages/wallet.vue]: change to subscribeWS and unsubscribeWS

**Pull request merged**